### DISCUSSION TITLE
CallerArgumentExpression

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#5061] [Q&A]
I've seen discussion about this and assumed it was implemented but I can't seen to figure out how to make it work.
What Am I doing something wrong?
```
#nullable enable

using System;
using System.Runtime.CompilerServices;

namespace ConsoleApp2
{
    class Program
    {
        static void Main(string[] args)
        {
            Write1("Hello" + " " + "World");
        }

        public static void Write1(string Value, [CallerArgumentExpression("Value")] string? Expression = default)
        {
            Console.WriteLine($@"{Value} = {Expression}");
        }
    }
}
```

Also, is it correct that I must use a literal string for CallerArgumentExpression and I can't use ```nameof(Value)``` like this:
```
        public static void Write2(string Value, [CallerArgumentExpression(nameof(Value)] string? Expression = default)
        {
            Console.WriteLine($@"{Value} = {Expression}");
        }
```

**Comment by ZacharyPatten** [comment]
See: https://github.com/dotnet/roslyn/blob/main/docs/Language%20Feature%20Status.md

`Caller expression attribute` -> "Merged into 17.0p3"

Make sure you are using a version it should be working in (the attribute was added but the compiler work was still pending).

**Reply by HaloFour** [reply]
I think the question is more about the expanded use of `nameof` to refer to method parameters, which is sitting in the "C# Next" working set.
